brave/brave-core
# Show download item origin URL.

Fixes brave/brave-browser#1269

* Subclasses DownloadItemView to draw the origin URL and the lock icon, updates accessible name with origin URL;
* Adds the lock icon to vector icons;
* Extends DownloadItemModel to obtain origin URL text and update tool tip text with origin URL;
* Adds unit tests for BraveDownloadItemModel.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Automated Test Plan:

`npm run test -- brave_unit_tests --filter=BraveDownloadItemModelTest.*`

## Test Plan:

1. Download from a secure site:
- Navigate to a secure (https) site, e.g. https://www.google.com.
- Right-click on the google logo and choose "Save image as...".
- Select destination if asked.
- Observe download item appear in the download shelf on the bottom of the browser.
- Verify that the download item shows the origin URL (https://www.google.com).
- Point mouse to the download item and wait for the tool tip to appear.
- Verify that the tool tip contains the origin URL.

2. Download from a non-secure site:
- Navigate to a non-secure (http) site, e.g. http://jonvilma.com/space.html.
- Right-click on any image and choose "Save image as...".
- Select destination if asked.
- Observe download item appear in the download shelf on the bottom of the browser.
- Verify that the download item shows the unlocked lock icon and the origin URL (http://jonvilma.com).
- Point mouse to the download item and wait for the tool tip to appear.
- Verify that the tool tip contains words "Not secure" followed by the origin URL.

3. Download from a file://:
- Drag and drop a text file onto the browser.
- Right-click on the content and choose "Save as...".
- Select destination if asked.
- Observe download item appear in the download shelf on the bottom of the browser.
- Verify that the download item shows the unlocked lock icon and the origin URL is `file:///`
- Point mouse to the download item and wait for the tool tip to appear.
- Verify that the tool tip contains words "Not secure file:///".

4. Download from ftp:
- Navigate to an ftp site, e.g: ftp://ftp.openssl.org/source/
- Right-click on a file (e.g. README) and choose "Save link as...".
- Select destination if asked.
- Observe download item appear in the download shelf on the bottom of the browser.
- Verify that the download item shows the unlocked lock icon and the origin URL (ftp://ftp.openssl.org).
- Point mouse to the download item and wait for the tool tip to appear.
- Verify that the tool tip contains words "Not secure" followed by the origin URL.

## Reviewer Checklist:
